src/utils.sh

```diff
@@ -116,3 +116,47 @@ log_debug() {
 }
 
 export -f log_init log_error log_info log_warn log_debug
+
+#
+ensure_sudo() {
+    if [[ $EUID -ne 0 ]];
+    then
+        echo "$0 is not running as root. "
+        sudo "$0"
+        exit $?
+    fi
+}
+
+export -f ensure_sudo
+
+#
+source /etc/os-release
+prepare_apt() {
+
+    case $ID in
+        ubuntu)
+            if [ $VERSION_ID != "18.04" ];
+            then
+                log_error "PREPARE_APT- OS VERSION: '%s' is not supported." $VERSION_ID
+                exit 1
+            fi
+            platform="ubuntu/18.04"
+            ;;
+
+        raspbian)
+            platform="debian/stretch"
+            ;;
+    esac
```

Comment:
the tier1 os check could save the platform for further use in the script so won't need to manipulate /etc/os-release multiple times?